src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -492,10 +492,17 @@ protected SimpleDateFormat initialValue() {
 
   private boolean isAuthenticationFailure(String code) {
     return "Core.LoginFailed".equals(code)
+        || "Core.FailedToAuthenticateWithOTDS".equals(code)
         || "AuthenticationService.Application.AuthenticationFailed"
         .equals(code);
   }
 
+  private boolean isPermissionsFailure(
+      SOAPFaultException soapFaultException) {
+    return "DocMan.PermissionsError".equals(
+        soapFaultException.getFault().getFaultCodeAsQName().getLocalPart());
```

Comment:
I hadn't really noticed it before, but we're assuming this call chain never fails, including a NPE. Is that OK?